python/cudf/cudf/core/series.py

```diff
@@ -2207,7 +2207,7 @@ def hash_values(self):
         """
         from cudf.core.column import numerical
 
-        return Series(numerical.column_hash_values(self._column))
+        return Series(numerical.column_hash_values(self._column)).values
```

Comment:
Not sure if there is proper dask_cudf test coverage to catch this, but we may need to modify cudf's `hash_object_dispatch` dispatch definition [which assumes `cudf.Series` output](https://github.com/rapidsai/cudf/blob/c4c2d9b04321f2a6acbdde5e6a4fde646fe08edf/python/dask_cudf/dask_cudf/backends.py#L71)

Reply:
Looks like the `dask_cudf` join/merge testing (which uses the hash dispatch functions) are working fine.  However, I did find a separate bug in `dask_cudf/backends.py` - Perhaps you can include [these changes](https://github.com/kkraus14/cudf/compare/align_index_setitem...rjzamora:dask_cudf_backend_tweaks) in this PR (otherwise, I can certainly submit them separately).